pkg/network/node/iptables.go

```diff
@@ -297,11 +297,16 @@ func (n *NodeIPTables) getNodeIPTablesChains() []Chain {
 
 func (n *NodeIPTables) ensureEgressIPRules(egressIP, mark string) error {
 	for _, cidr := range n.clusterNetworkCIDR {
-		err := execIPTablesWithRetry(func() error {
+		if err := execIPTablesWithRetry(func() error {
 			_, err := n.ipt.EnsureRule(iptables.Prepend, iptables.TableNAT, iptables.Chain("OPENSHIFT-MASQUERADE"), "-s", cidr, "-m", "mark", "--mark", mark, "-j", "SNAT", "--to-source", egressIP)
 			return err
-		})
-		if err != nil {
+		}); err != nil {
+			return err
+		}
+		if err := execIPTablesWithRetry(func() error {
+			_, err := n.ipt.EnsureRule(iptables.Prepend, iptables.TableNAT, iptables.Chain("OPENSHIFT-MASQUERADE"), "-s", cidr, "-d", cidr, "-m", "mark", "--mark", mark, "-j", "MASQUERADE")
```

Comment:
you should be able to do

    execIPTablesWithRetry(func() error {
            EnsureRule(...)
            EnsureRule(...)
    })

(with appropriate error checking) rather than

    execIPTablesWithRetry(func() error {
            EnsureRule(...)
    })
    execIPTablesWithRetry(func() error {
            EnsureRule(...)
    })